### PR TITLE
Add and use internal X509_add_certs_new(), which simplifies matters

### PR DESCRIPTION
**Pull request by DDvO**
This has been extracted from #13070.

### CODE REVIEW

**Comment by DDvO** [comment]
Rebased to solve a merge conflict.

**Comment by DDvO** [comment]
Rebased to solve merge conflicts

**Comment by DDvO** [comment]
Can we please handle this soon?
It keeps suffering from merge conflicts,
which I just solved for the third time.

**Comment by t8m** [comment]
According to #12964  it should be named ossl_x509_add_certs_new().

**Comment by DDvO** [comment]
Done.
Consequently, also renamed `X509_add_cert_new()` to `ossl_x509_add_cert_new()`.

File: crypto/ocsp/ocsp_vfy.c [hunk 1]
```diff
@@ -118,10 +118,6 @@ int OCSP_basic_verify(OCSP_BASICRESP *bs, STACK_OF(X509) *certs,
                 goto end;
             if (!X509_add_certs(untrusted, certs, X509_ADD_FLAG_DEFAULT))
                 goto end;
-        } else if (certs != NULL) {
-            untrusted = certs;
-        } else {
-            untrusted = bs->certs;
```
Comment (t8m) [inline crypto/ocsp/ocsp_vfy.c:124]: How is this change related?

Comment (DDvO) [reply crypto/ocsp/ocsp_vfy.c:124]: This was poor man's concatenation of two cert lists in case either was empty.

Comment (t8m) [reply crypto/ocsp/ocsp_vfy.c:124]: So it does not matter that untrusted will be NULL in case flags has OCSP_NOCHAIN  set?

Comment (DDvO) [reply crypto/ocsp/ocsp_vfy.c:124]: The `untrusted` list gets just passed to `X509_STORE_CTX_init()` as last parameter, which allows NULL.

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
> The renaming is not necessary since the old name was not exported from the library.

Yeah, the renaming was just for better adherence to the internal naming conventions.

**Comment by DDvO** [comment]
Merged - thanks @t8m
